### DISCUSSION TITLE
Align multiple cursors with whitespace

### DISCUSSION THREAD

**Discussion by u1f98e** [helix-editor/helix#12186] [Ideas] [answered]
I run into a need to align things on multiple lines fairly frequently. For example, basic tables:
```c++
// Parameter   Min    Max
// x           0      500
// x_2         50     1000
// x_3         1000   20000 
...
```
Editing large or complex tables like this manually becomes terrible pretty quickly, and I often need to jump to another editor to take care of formatting. Other cases include aligning comments, declarations, or assignments. Those might get taken care of by a formatter, though a formatter isn't always available/doesn't always have the behavior I need.

Helix's multiple selection model lends itself to a simple way to do this: For each first selection on a line, insert spaces before the beginning and end of the first selections on other lines until all beginnings and ends are aligned, and repeat for following selections. This [vscode extension](https://marketplace.visualstudio.com/items?itemName=yo1dog.cursor-align) provides similar functionality.

I don't believe there's a command to do this, if there's a simple motion someone knows I'd love to hear it. If not, I'd like to make a PR for an `:align` command which does this. I believe it's a basic and useful enough feature that it would make sense as a built in rather than an eventual plugin, but wanted to get feedback first.

**Comment by the-mikedavis** [comment, marked as answer]
I think what you're looking for is already covered by the `align_selections` command bound to `&` by default

**Reply by u1f98e** [reply]
That's it, don't know why I missed it earlier. Thanks!
